cluster/resolver/raft.go

```diff
@@ -66,7 +66,7 @@ func (a *raft) ServerAddr(id raftImpl.ServerID) (raftImpl.ServerAddress, error)
 	defer a.nodesLock.Unlock()
 	if addr == "" {
 		a.notResolvedNodes[id] = struct{}{}
-		return raftImpl.ServerAddress(invalidAddr), nil
+		return "", fmt.Errorf("could not resolve server id %s", id)
```

Comment:
> this PR removes the invalid ip returns to avoid retrying for invalid IP and instead return error which forcing RAFT to fall back to old IP and try to resolve new IP.

We can't do that, the problem is that raft will shoot replication to the old IP (which might be a different cluster). That opens the situation for example where

cluster 1
node1 -> 1.1.1.1
node -> 2.2.2.2
node2 dies, and doesn't restart (for whichever reason) so it never re-joins memberlist with a new IP
-> node1 will hammer 2.2.2.2 with request, and the IP might get re-allocated to a different cluster

Unless we're forking raft to add labels and filter at the raft level traffic between different cluster i'm not comfortable merging this as it is.

Reply:
Or an alternative could be that on a node becoming unhealthy in memberlist we also remove it from RAFT

Reply:
that's why we force the node to exit and if the node is leader it will take care of removing the node and re-add it

Reply:
basically and raft add/remove shall happen only on the leader
the NotifyConflict() take cars of both
if the `leader` will `remove/add` nodes and  remove called to make sure we stop replication
if the node itself is the one got conflicted it will kill itself `exit(1)`